flight/Modules/ManualControl/transmitter_control.c

```diff
@@ -1201,13 +1201,19 @@ bool validInputRange(int16_t min, int16_t max, uint16_t value, uint16_t offset)
  */
 static void applyDeadband(float *value, float deadband)
 {
-	if (fabsf(*value) < deadband)
+	if (deadband < 0.00001f) return;
+
+	if (fabsf(*value) < deadband) {
 		*value = 0.0f;
-	else
-		if (*value > 0.0f)
+	} else {
+		if (*value > 0.0f) {
 			*value -= deadband;
-		else
+		} else {
 			*value += deadband;
+		}
+
+		*value /= 1 - deadband;
```

Comment:
Also, we need to make sure that ```deadband``` is strictly ```>0``` in order to avoid any possible ```NaN``` contamination. If we don't want to do this check every cycle, this should be easy to do at line https://github.com/mlyle/TauLabs/blob/mpl-deadbandrange/flight/Modules/ManualControl/transmitter_control.c#L169.